akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -722,29 +724,35 @@ class ShardRegion(
         }
     }
 
-  def getShard(id: ShardId): Option[ActorRef] = shards.get(id).orElse(
-    entityProps match {
-      case Some(props) if !shardsByRef.values.exists(_ == id) ⇒
-        log.debug("Starting shard [{}] in region", id)
-
-        val name = URLEncoder.encode(id, "utf-8")
-        val shard = context.watch(context.actorOf(
-          Shard.props(
-            typeName,
-            id,
-            props,
-            settings,
-            extractEntityId,
-            extractShardId,
-            handOffStopMessage).withDispatcher(context.props.dispatcher),
-          name))
-        shardsByRef = shardsByRef.updated(shard, id)
-        None
-      case Some(props) ⇒
-        None
-      case None ⇒
-        throw new IllegalStateException("Shard must not be allocated to a proxy only ShardRegion")
-    })
+  def getShard(id: ShardId): Option[ActorRef] = startingShards.contains(id) match {
+    case true ⇒ None
```

Comment:
You're right, maybe it's more common to do the regular if-else. I'll change that.